### PR TITLE
rest-backend: fixes

### PR DESCRIPTION
**Pull request by Namoshek**
# Purpose

Fixes some issues and #1151 features.

# Checklist

- [x] commit messages are fine (with references to issues)
- [x] I ran all tests and everything went fine

@markus2330 please review my pull request

### CODE REVIEW

**Comment by Namoshek** [comment]
jenkins build homepage please

**Comment by Namoshek** [comment]
jenkins build homepage please

**Comment by Namoshek** [comment]
@markus2330 please check the black screen issue and look through the commits if it looks fine to you

**Comment by markus2330** [comment]
Black screen issue is still there :cry:  But it seems like there was a change. Did you try to use firefox-esr from jessie?

File: scripts/update-snippet-repository [hunk 1]
```diff
@@ -2,19 +2,22 @@
 
 SNIPPET_REPO=/var/lib/jenkins/snippets
 SNIPPET_ROOT=system/configs
-SNIPPET_EXPORT_FILENAME=snippets
 LOCK_FILE=/run/elektra-update-snippet-repository.lock
+KEY_SLASH_REPLACED=$(echo "$ELEKTRA_REST_KEY" | tr '/' '_')
 
 export_snippets() {
-  # snippets.dump does not need to be exported, it is the source file
-  kdb export $SNIPPET_ROOT ni > $SNIPPET_EXPORT_FILENAME.ni.ini
```
Comment (markus2330) [inline scripts/update-snippet-repository]: The idea was to have both: everything in a file and every file separated. Is there a specific reason why you removed the everything in a file?

File: src/tools/rest-backend/benchmarks/benchmark_kdbrest_mysql.cpp [hunk 2]
```diff
@@ -63,7 +63,7 @@ void clearDatabase (sql::Connection * con)
 		"`author_id` INT NOT NULL, `organization` VARCHAR(255) NOT NULL, `application` VARCHAR(255) NOT NULL, "
 		"`scope` VARCHAR(255) NOT NULL, `slug` VARCHAR(255) NOT NULL, `title` VARCHAR(255) NOT NULL, "
 		"`description` TEXT NOT NULL, `configuration` TEXT NOT NULL, `plugin` VARCHAR(100) NOT NULL, "
-		"`views` INT NOT NULL DEFAULT '0', `createdat` INT NOT NULL, PRIMARY KEY (`id`), "
+		"`createdat` INT NOT NULL, PRIMARY KEY (`id`), "
```
Comment (markus2330) [inline src/tools/rest-backend/benchmarks/benchmark_kdbrest_mysql.cpp]: Did you rerun the benchmarks?

File: src/tools/rest-frontend/Gruntfile.js [hunk 3]
```diff
@@ -219,13 +219,26 @@ module.exports = function (grunt) {
...
                 src: 'resources/assets/js/application.js',
-                dest: 'public/assets/js/application.js',
+                dest: 'resources/application.js.tmp',
                 options: {
                     banner: dstFileBanner,
                     watch: true
                 }
             }
         },
+        uglify: {
```
Comment (markus2330) [inline src/tools/rest-frontend/Gruntfile.js]: Did you check if the compressed file is smaller, too?

**Comment by Namoshek** [comment]
Sure, both version 45.4.0 and 45.6.0 (current stable).

File: src/tools/rest-frontend/public/assets/translations/en.json [hunk 4]
```diff
@@ -328,8 +329,8 @@
     "APP.HOME.LABEL.TOOL.OTHERS.HEADLINE": "Other Applications",
     "APP.HOME.LABEL.TOOL.QTGUI.DETAILS": "Elektra offers a GUI based on Qt Quick. It simplifies common tasks and offers easy access to the key database itself. A description for the GUI tool can be found <a href=\"/tools/qt-gui\">here</a>.",
     "APP.HOME.LABEL.TOOL.QTGUI.HEADLINE": "Graphical User Interface",
-    "APP.HOME.LABEL.TOOL.WEBGUI.DETAILS": "The Web GUI allows for remote configuration of systems. Besides the configuration of single systems, it also features cluster configuration.",
-    "APP.HOME.LABEL.TOOL.WEBGUI.HEADLINE": "Web GUI",
+    "APP.HOME.LABEL.TOOL.WEBGUI.DETAILS": "The Web GUI will allow for remote configuration of systems. Besides the configuration of single systems, it will also feature cluster configuration.",
```
Comment (markus2330) [inline src/tools/rest-frontend/public/assets/translations/en.json]: Info (not specific to this line): Please comment out the web UI or write (to be released)

File: src/tools/rest-frontend/resources/assets/html/index.html.in [hunk 5]
```diff
@@ -71,12 +71,12 @@
...
         <div ui-view></div>
 
-        <!-- Application JavaScript -->
-        <script src="assets/js/application.js"></script>
-
         <!-- AsciiNema JS -->
         <script src="vendor/asciinema/asciinema-player.js"></script>
 
+        <!-- Application JavaScript -->
+        <script src="assets/js/application.js"></script>
```
Comment (markus2330) [inline src/tools/rest-frontend/resources/assets/html/index.html.in]: Indeed a strange bug.

File: src/tools/rest-frontend/Gruntfile.js (hunk 3 above)
Comment (Namoshek) [inline src/tools/rest-frontend/Gruntfile.js]: I said it in the meeting: 3mb -> 1.5mb.

File: src/tools/rest-frontend/public/assets/translations/en.json (hunk 4 above)
Comment (Namoshek) [inline src/tools/rest-frontend/public/assets/translations/en.json]: That's why I wrote `Upcoming` one line below in the Web GUI title and changed the verbs to future...

**Comment by markus2330** [comment]
Thank you! Great job!

**Pull request merged**